src/hooks/useAllServices.ts

```diff
@@ -95,105 +105,119 @@ const filterAllServicesSections = (allServicesLinks: AllServicesSection[], filte
     return acc;
   }, []);
 };
+
+const findNavItems = (
+  items: (string | AllServicesLink | AllServicesGroup)[] = [],
+  availableLinks: { id?: string; title?: string; items: AvailableLinks }[]
+): (AllServicesLink | AllServicesGroup)[] =>
+  items
+    .map((item) => {
+      if (isAllServicesGroup(item)) {
+        return {
+          ...item,
+          links: findNavItems(item.links, availableLinks),
+        };
+      } else if (isAllServicesLink(item)) {
+        return item;
+      }
+      const [bundle, nav] = (item as string).split('.');
+      const currBundle = availableLinks.find(({ id }) => id === bundle)?.items || {};
+      return Object.values(currBundle).find(({ id }) => id === nav);
+    })
+    .filter(Boolean) as (AllServicesLink | AllServicesGroup)[];
+
 const useAllServices = () => {
-  const [{ availableLinks, ready, error }, setState] = useState<{
+  const [{ ready, error, availableSections }, setState] = useState<{
     error: boolean;
     ready: boolean;
-    availableLinks: NavItem[];
+    availableLinks: BundleNav[];
+    availableSections: AllServicesSection[];
   }>({
     ready: false,
     availableLinks: [],
     error: false,
+    availableSections: [],
   });
   const isMounted = useRef(false);
   const [filterValue, setFilterValue] = useState('');
   // TODO: move constant once the AppFilter is fully replaced
   const bundles = requiredBundles;
+  const fetchNavitation = useCallback(
+    () =>
+      Promise.all(
+        bundles.map((fragment) =>
+          axios
+            .get<BundleNav>(`${getChromeStaticPathname('navigation')}/${fragment}-navigation.json?ts=${Date.now()}`)
+            .catch(() => axios.get<BundleNavigation>(`${isBeta() ? '/beta' : ''}/config/chrome/${fragment}-navigation.json?ts=${Date.now()}`))
+            .then(handleBundleResponse)
+            .catch((err) => {
+              console.error('Unable to load appfilter bundle', err, fragment);
+              return [];
+            })
+        )
+      ),
+    []
+  );
+  const fetchSections = useCallback(
+    async (): Promise<
+      (Omit<AllServicesSection, 'links'> & {
+        links: (string | AllServicesLink | AllServicesGroup)[];
+      })[]
+    > => (await axios.get(`${getChromeStaticPathname('services')}/services.json`)).data,
```

Comment:
The type should be part of the `axios.get` generic. It will be safer for the future as we get the type directly as the axios response payload instead of forcing the callback return type.

```ts
  const fetchSections = useCallback(
    async () =>
      (
        await axios.get<
          (Omit<AllServicesSection, 'links'> & {
            links: (string | AllServicesLink | AllServicesGroup)[];
          })[]
        >(`${getChromeStaticPathname('services')}/services.json`)
      ).data,
    []
  );
```